AutoFlow_OmicsDataHandling/INCA_script_generator.py

```diff
@@ -0,0 +1,1232 @@
+                                    molecule_index
+                                ][mapping_cnt]
+                                + " "
+                            )
+                elif type(atomMapping_molecules_elements) is list:
+                    for mapping_cnt, mapping in enumerate(
+                        atomMapping_molecules_elements
+                    ):
+                        # to remove the final whitespace
+                        if (mapping_cnt + 1) == len(
+                            atomMapping_molecules_mapping[0]
+                        ):
+                            rxn_equation += (
+                                mapping
+                                + str(mapping_cnt + 1)
+                                + ":"
+                                + atomMapping_molecules_mapping[0][mapping_cnt]
+                            )
+                        else:
+                            rxn_equation += (
+                                mapping
+                                + str(mapping_cnt + 1)
+                                + ":"
+                                + atomMapping_molecules_mapping[0][mapping_cnt]
+                                + " "
+                            )
+                # close the brackets for mapping
+                rxn_equation += ")"
+
+            # Exceptions that are in the model but not tracked are added here
+            else:
+                rxn_equation += (
+                    str(model_molecules_stoichiometry[molecule_cnt])
+                    + "*"
+                    + molecule
+                )
+                duplicates_not_done = False
+
+    return rxn_equation
+
+
+def add_reactions_to_script(modelReaction_data_I, atomMappingReactions_data_I):
+    """
+    Translates the model and adds mapping using reaction_mapping()
+
+    Parameters:
+        modelReaction_data_I: pre-processed modelReaction_data_I input data
+        atomMappingReactions_data_I: pre-processed atomMappingReactions_data_I
+            input data
+
+    Returns:
+        mat_script: Extention to the MATLAB script under construction
+        model_rxn_ids_exp: List of reaction IDs used for the model
+
+    """
+    mat_script = "r = reaction({... % define reactions\n"
+
+    # this is the teporary biomass function
+    # biomass_INCA_iJS2012
+    a = "0.176*phe_DASH_L_c + 0.443*mlthf_c + 0.34*oaa_c + 0.326*lys_DASH_L_c + 33.247*atp_c + 0.205*ser_DASH_L_c + 0.129*g3p_c + 0.131*tyr_DASH_L_c + 0.051*pep_c + 0.146*met_DASH_L_c + 0.205*g6p_c + 0.087*akg_c + 0.25*glu_DASH_L_c + 0.25*gln_DASH_L_c + 0.754*r5p_c + 0.071*f6p_c + 0.083*pyr_c + 0.582*gly_c + 0.241*thr_DASH_L_c + 0.229*asp_DASH_L_c + 5.363*nadph_c + 0.087*cys_DASH_L_c + 0.619*3pg_c + 0.402*val_DASH_L_c + 0.488*ala_DASH_L_c + 0.276*ile_DASH_L_c + 0.229*asn_DASH_L_c + 0.09*his_DASH_L_c + 0.428*leu_DASH_L_c + 2.51*accoa_c + 0.281*arg_DASH_L_c + 0.21*pro_DASH_L_c + 0.054*trp_DASH_L_c -> 1.455*nadh_c + 39.68*Biomass_c" # noqa E501
```

Comment:
I recommend we do something that is a lot more extensible than the hard coding of the biomass function as I did in my original scripts.  

two thoughts come to mind:
1. We could add it in as a parameter to the method.
2. We could add it to the .csv file so that it is added in automatically (given this is possible)

Reply:
I would add it as a parameter for now and make sure to include a catalogue for different organisms later on for actual users

Reply:
Added the biomass function to the example notebook for now